### PR TITLE
Added an --extra_ports option.

### PR DESCRIPTION
**Pull request by mindhog**
Added an option allowing users to specify additional port mappings to the
application container.

@rbuskens please review.

### CODE REVIEW

File: appstart/cli/parsing.py [hunk 1]
```diff
@@ -22,6 +22,30 @@
...
+        result = {}
+        try:
+            for val in vals:
+                try:
+                    host_port, container_port = val.split(':')
+                except ValueError:
+                    host_port = container_port = val
+                result[int(container_port)] = int(host_port)
+        except ValueError:
+            parser.error('bad value for option {0}.  Expected a comma '
```
Comment (rbuskens) [inline appstart/cli/parsing.py]: 'b' --> 'B'.

File: appstart/cli/parsing.py [hunk 2]
```diff
@@ -202,3 +226,15 @@ def add_appstart_args(parser):
...
     parser.set_defaults(clear_datastore=False)
+
+    parser.add_argument('--extra_ports',
+                        action=StorePortMapAction,
+                        dest='extra_ports',
+                        help='A comma separated map of '
+                        'host_port:container_port pairs defining the mapping '
+                        'from host ports to container ports on the '
+                        'application container for an arbitrary set of ports.'
+                        'If a single value is provided instead of a colon '
```
Comment (rbuskens) [inline appstart/cli/parsing.py:237]: 'colon separated' --> 'colon-separated'

File: tests/system_tests/services_test_app.py [hunk 3]
```diff
@@ -79,9 +84,32 @@ def get(self):
...
+    s.listen(5)
+    while True:
+        c, addr = s.accept()
+        data = c.recv(1024)
+        c.send(data)
+        c.close()
+
+
+class OpenPort(webapp2.RequestHandler):
+    """Open port 1000."""
```
Comment (rbuskens) [inline tests/system_tests/services_test_app.py:102]: Move this comment to socket_thread()?

File: appstart/cli/parsing.py (hunk 1 above)
Comment (mindhog) [inline appstart/cli/parsing.py]: Done.

File: appstart/cli/parsing.py (hunk 2 above)
Comment (mindhog) [inline appstart/cli/parsing.py:237]: Done.

File: tests/system_tests/services_test_app.py (hunk 3 above)
Comment (mindhog) [inline tests/system_tests/services_test_app.py:102]: The "Open port 1000?"  It's intended for viewers of the web interface which is in the OpenPort class, so I want it there at minimum.  I'll add more detailed commentary to socket_thread(), though.

Comment (rbuskens) [inline tests/system_tests/services_test_app.py:102]: Why not pass 1000 into the socket_thread() call instead?

**Pull request merged**